src/AstronautsTable.js

```diff
@@ -1,38 +1,53 @@
-import React from 'react';
-import astronauts from './astronauts.json'
+import React from "react";
+import { useState, useEffect } from "react";
+import fetchJson from "./fetchJson";
 
-function AstronautsTable(){
-    const DisplayData=astronauts.people.map(
-        (info)=>{
-            return(
-                <tr>
-                    <td>{info.craft}</td>
-                    <td>{info.name}</td>
-                </tr>
-            )
-        }
-    )
+function AstronautsTable() {
+  const [error, setError] = useState(null);
+  const [astronauts, setAstronauts] = useState([]);
+  const [isLoaded, setIsLoaded] = useState(false);
 
- 
-    return(
-        <div>
-            <table class="table table-striped">
-                <thead>
-                    <tr>
-                    <th>Craft</th>
-                    <th>Name</th>
-                    </tr>
-                </thead>
-                <tbody>
-                 
-                    
-                    {DisplayData}
-                    
-                </tbody>
-            </table>
-             
-        </div>
-    )
- }
- 
- export default AstronautsTable;
\ No newline at end of file
+  useEffect(() => {
+    fetchJson().then(
+      (data) => {
+        setAstronauts(data);
+        setIsLoaded(true);
+      },
+      (error) => {
+        setError(error);
+        setIsLoaded(true);
+      }
+    );
+  }, []);
+
+  if (error) {
+    return <div>Error: {error.message}</div>;
+  } else if (!isLoaded) {
+    return <div>Loading, please wait...</div>;
+  } else {
+    const astronautRow = astronauts.people.map((astronaut) => {
+      return (
+        <tr>
```

Comment:
This could be an iterable row so 
```
.map(a => {
  <Astronaut name={a.name} etc... />
```

This will give you some work with passing props and importing other components.

Reply:
Done! Nice and neat